### PR TITLE
Add regtests for sigma mint spend with 'watchonly' address

### PR DESCRIPTION
**Pull request by thebevrishot**
add reg test following this card https://trello.com/c/tkaqvuh0/378-add-regtests-for-sigma-mint-spend-with-watchonly-address-t15d1

### CODE REVIEW

**Comment by nopslide** [comment]
This pull request **introduces 1 alert** when merging 2fe5dced2f6a164047e3a33310bd82fc0be773ac into 8b7d49ef63387ab7b813ea95844369fb08995ef4 - [view on LGTM.com](https://lgtm.com/projects/g/zcoinofficial/zcoin/rev/pr-2c00dd5c0bf8dc6c64060ba1a047c535556394f0)

**new alerts:**

* 1 for &#39;import \*&#39; may pollute namespace

---

This project has automated code review enabled, but doesn't use the <a href="https://github.com/apps/lgtm-com">LGTM GitHub App</a>. Migrate over by <a href="https://github.com/apps/lgtm-com/installations/new">installing the app</a>. Read about the <a href="https://lgtm.com/blog/announcing_github_apps_migration">benefits of migrating to GitHub Apps</a> in the blog.

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: qa/rpc-tests/sigma_spend_gettransaction.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+        self.nodes[0].generate(10)
+        self.sync_all()
+
+        # case 1: Spend many with watchonly address
+        spendto_wo_id = self.nodes[0].spendmany("", {watchonly_address: 1})
+        spendto_wo_tx = self.nodes[0].gettransaction(spendto_wo_id)
+
+        assert spendto_wo_tx['amount'] == Decimal('-1')
+        assert spendto_wo_tx['fee'] == Decimal('-0.05')
```
Comment (ultimaweapon) [inline qa/rpc-tests/sigma_spend_gettransaction.py]: I think we should not check fee with exactly value here due to this test is not a test for fee algorithm. We should check only if there is fee.

File: qa/rpc-tests/sigma_spend_gettransaction.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        assert isinstance(spendto_wo_tx['details'], list)
+        assert len(spendto_wo_tx['details']) == 1
+        assert spendto_wo_tx['details'][0]['involvesWatchonly']
+
+        # case 2: Spend many with watchonly address and valid address
+        spendto_wo_and_valid_id = self.nodes[0].spendmany("", {watchonly_address: 1, valid_address: 2})
+        spendto_wo_and_valid_tx = self.nodes[0].gettransaction(spendto_wo_and_valid_id)
+
+        assert spendto_wo_and_valid_tx['amount'] == Decimal('-1')
+        assert spendto_wo_and_valid_tx['fee'] == Decimal('-0.05')
```
Comment (ultimaweapon) [inline qa/rpc-tests/sigma_spend_gettransaction.py]: Same here.

**Comment by thebevrishot** [comment]
@ultimaweapon this file have been already allowed to execute.

**Comment by nopslide** [comment]
This pull request **introduces 1 alert** when merging 7711b96be517dce0b04e5e2ee55335fb33bfed86 into 8b7d49ef63387ab7b813ea95844369fb08995ef4 - [view on LGTM.com](https://lgtm.com/projects/g/zcoinofficial/zcoin/rev/pr-36e5d6efbe3b0c221bd8170fd48657652977ac21)

**new alerts:**

* 1 for &#39;import \*&#39; may pollute namespace

---

This project has automated code review enabled, but doesn't use the <a href="https://github.com/apps/lgtm-com">LGTM GitHub App</a>. Migrate over by <a href="https://github.com/apps/lgtm-com/installations/new">installing the app</a>. Read about the <a href="https://lgtm.com/blog/announcing_github_apps_migration">benefits of migrating to GitHub Apps</a> in the blog.

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Pull request merged**